fsspec/generic.py

```diff
@@ -136,7 +136,8 @@ def rsync(
         fs.cp(source_files, target_files, **kwargs)
     logger.debug(f"{len(to_delete)} files to delete")
     if delete_missing:
-        fs.rm(to_delete)
+        for file in to_delete:
+            fs.rm(file)
```

Comment:
No, we definitely don't want this - all rm() methods should be able to delete in batch and that's the right call here. 

(There is a separate rm_file method that requires a single path. )

The problem, instead, is in attempting to resolve the filesystem protocol from a list.